group_by.py

```diff
@@ -0,0 +1,7 @@
+def group_by(f, it):
+    dic={}
+    for i in it:
```

Comment:
Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing

'f', 'it' 'dic' 'i' aren't saying anything, imagine running into this as part of a 500 lines of code, you have to practice and to get used to giving explanatory names